# U.S. Medical Insurance Costs

# Basic Queries
- Find out the average age of the patients in the dataset.
- Analyze where a majority of the individuals are from.
- Figure out what the average age is for someone who has at least one child in this dataset.

In [1]:
# US Medical Insurance Cost
import csv

In [2]:
ages = []
with open('insurance.csv') as average_age_file:
    average_age = csv.DictReader(average_age_file)
    for row in average_age:
        ages.append(row['age'])

In [3]:
print(len(ages))

1338


# Getting the average age of the group

In [4]:
total_age = 0
for age in ages:
    total_age = total_age + float(age)

In [5]:
print(total_age)

52459.0


In [6]:
# now lets get the average
average = total_age / len(ages)
print(average)

39.20702541106129


The average age of the group is 39.2 years

# Finding out where the majority of the individuals are from.

In [7]:
# creating a regions list 
regions = []
with open('insurance.csv') as regions_file:
    region_reader = csv.DictReader(regions_file)
    for row in region_reader:
        regions.append(row['region'])

In [8]:
# creating 4 list for the 4 regions
s_west = []
s_east = []
n_west = []
n_east = []
for region in regions:
    if region == 'southwest':
        s_west.append(region)
    elif region == 'southeast':
        s_east.append(region)
    elif region == 'northwest':
        n_west.append(region)
    else:
        n_east.append(region)
        
print("There are " + str(len(s_west)) + " from the southwest region.")
print("There are " + str(len(s_east)) + " from the southeast region.")
print("There are " + str(len(n_west)) + " from the northwest region.")
print("There are " + str(len(n_east)) + " from the southeast region.")

There are 325 from the southwest region.
There are 364 from the southeast region.
There are 325 from the northwest region.
There are 324 from the southeast region.


# Investigating the average age of someone with at least 1 child

In [9]:
# finding out the average age of someone with atleast a child
adultAges = []
no_of_children = []
with open('insurance.csv') as age_for_ppl_with_kids_file:
    their_avg_age = csv.DictReader(age_for_ppl_with_kids_file)
    for row in their_avg_age:
        adultAges.append(row['age'])
        no_of_children.append(row['children'])


In [10]:
age_and_no_of_children = list(zip(adultAges, no_of_children))
# print(age_and_no_of_children)
tot = 0
count = 0
for num in age_and_no_of_children:
    if float(num[1]) >= 1:
        tot += float(num[0])
        count += 1
print('Total age of people with one child and more '+ str(tot))
print('Total number of people with one child and more '+ str(count))
# getting the average age
average_age = tot / count
print('And the average age is: ' + str(average_age))

Total age of people with one child and more 30392.0
Total number of people with one child and more 764
And the average age is: 39.78010471204188


# Below is a function to get the youngest and the oldest in the group.

In [31]:
# Getting the youngest and the oldest
def youngest_and_oldest():
    min_age = 100000.0
    max_age = float("-inf")
    for age in ages:
        if float(age) > max_age:
            max_age = float(age)
        if float(age) < min_age:
            min_age = float(age)
    return 'The youngest in this group is ' + str(min_age) + ' years old and the oldest is ' + str(max_age) + ' years old.' 
print(youngest_and_oldest())

The youngest in this group is 18.0 years old and the oldest is 64.0 years old.


# Function to get the number of smokers and non-smokers.

In [44]:
# Difference of cost between smokers and non-smokers
def smoker_and_non_smokers():
    smoking = []
    smokers = []
    non_smokers = []
    with open('insurance.csv') as lifestyle_file:
        lifestyle = csv.DictReader(lifestyle_file)
        for row in lifestyle:
            smoking.append(row['smoker'])
        for indv in smoking:
            if indv == 'yes':
                smokers.append(indv)
            else:
                non_smokers.append(indv)
    return 'Smokers are '+ str(len(smokers))+ ' and non-smokers are ' + str(len(non_smokers))
print(smoker_and_non_smokers())

Smokers are 274 and non-smokers are 1064


# Final comments.
- There are 1338 individuals in the this dataset with an average group of 39 years old.
- 764 people have a child or more and their average age is 39.7 years old.
- The youngest person in this dataset is 18 years old and the oldest is 64 years old.
- There are 274 smokers and 1,064 non-smokers in this dataset.
- Most people in this dataset are from the southeast reagion with 364.

- P.S. I know there's alot of repetition in this analysis(code wise) and I went for functionality rather than efficiency. Working on writing cleaner and more concise code. 
